pycqed/utilities/general.py

```diff
@@ -800,3 +800,65 @@ def find_symmetry_index(data):
         data_filtered = data[np.int(iflip-span):np.int(iflip+span+1)]
         corr.append((data_filtered*data_filtered[::-1]).sum())
     return np.argmax(corr), corr
+
+
+def raise_warning_image(destination_path, warning_image_path=None):
+    """
+    Copy the image specified by warning_image_path to the folder specified by
+    destination_path.
+    :param destination_path: folder where the warning image is to be copied
+    :param image_path: full path, including image name and extension, to
+        the warning image to be copied. If None, assumes WARNING.png exists in
+        the module folder.
+    :return:
+    """
+    if 'WARNING.png' not in os.listdir(destination_path):
+        import shutil
+        if warning_image_path is None:
+            warning_image_path = os.path.abspath(sys.modules[__name__].__file__)
+            warning_image_path = os.path.split(warning_image_path)[0]
+            warning_image_path = os.path.abspath(os.path.join(
+                warning_image_path, 'WARNING.png'))
+
+        destination = os.path.abspath(os.path.join(destination_path,
+                                                   'WARNING.png'))
+        shutil.copy2(warning_image_path, destination)
+
+
+def write_warning_message_to_text_file(destination_path, message, filename=None):
+    """
+    Write a warning message to a text file.
+    :param destination_path: folder to the text file. If file does not yet
+        exist, it will be created.
+    :param message: string with the message to be written into the text file.
+    :param filename: string with name of the warning message file without
+        extension. If None, uses 'warning_message'. If text file does not exist,
+        it will be created. It text file already exists, the message wil be
+        appended.
+    :return:
+    """
+    if filename is None:
+        filename = 'warning_message'
+    filename += '.txt'
```

Comment:
I would do this only if the filename does not have an extension yet

Reply:
Changed now